spring-cloud-contract-verifier/src/main/groovy/org/springframework/cloud/contract/verifier/builder/JsonBodyVerificationBuilder.java

```diff
@@ -102,14 +103,21 @@ Object addJsonResponseBodyCheck(BlockBuilder bb, Object convertedResponseBody,
 		Object copiedBody = cloneBody(convertedResponseBody);
 		convertedResponseBody = JsonToJsonPathsConverter
 				.removeMatchingJsonPaths(convertedResponseBody, bodyMatchers);
+
+		// If it was a map or list where all elements covered by matchers - json paths
+		// should not include empty check
+		boolean includeEmptyCheck = !mapOrListBodyReducedToEmpty(copiedBody,
```

Comment:
This covers the case where the body comes in with fields but they are covered by the matchers so the `convertedResponseBody` is empty. In this case we do not want to add the `isEmpty` check because it is not actually empty, its just "realized" as empty to appease the assertion generation gods.

Reply:
Welcome to the world of assertion gods ;)